File: binary_search/find_minimum_in_rotated_sorted_array/js/find_minimum_in_rotated_sorted_array.js
const nums = [3, 4, 5, 1, 2];

const findMin = (nums) => {
  let [low, high] = [0, nums.length-1];
  let mid = (low + high) >> 1;
  let answer = nums[mid];

  while(low <= high) {
    mid = (low + high) >> 1;
    nums[high] > nums[mid] ? high = mid : low = mid+1;
    answer = Math.min(answer, nums[mid]);
  }
  return answer;
}

console.log(findMin(nums));

File: binary_search/find_minimum_in_rotated_sorted_array/cpp/find_minimum_in_rotated_sorted_array.cpp
#include<iostream>
#include<vector>

using std::vector;

// Time O(log(n))
// Space O(1)

class Solution {
public:
    static int findMin(const vector<int>& nums) {
        int low = 0;
        int high = nums.size()-1;
        int answer = nums[high / 2];

        while(low <= high) {
            int mid = (low + high) >> 1;
            nums[high] > nums[mid] ? high = mid : low = mid+1;
            answer = std::min(answer, nums[mid]);
        }
        return answer;
    }
};

int main() {
    vector<int> nums = { 3, 4, 5, 1, 2 };

    int answer = Solution::findMin(nums);
    std::cout << answer << '\n';
}
